### PR TITLE
feat(changeStream): allow resuming on getMore errors

### PR DESCRIPTION
**Pull request by daprahamian**
Fixes NODE-1462

Requires mongodb-js/mongodb-core#309

### CODE REVIEW

File: lib/change_stream.js [hunk 1]
```diff
@@ -312,11 +313,30 @@ const RESUMABLE_ERROR_CODES = new Set([
...
+
+  if (GET_MORE_NON_RESUMABLE_CODES.has(error.code)) {
+    return false;
+  }
+
+  return true;
+}
 
 function isResumableError(error) {
   // TODO: Need a way to check if error is
```
Comment (kvwalker) [inline lib/change_stream.js]: This TODO can be removed now, right?

Comment (daprahamian) [reply lib/change_stream.js]: yes it can!

**Comment by daprahamian** [comment]
Ok, I pushed an update. Apparently I was working off of an older version of the spec, and we shouldn't be resuming on aggregates, only getMores. So I updated the check for resuming, and I wrote new tests to reflect that.

**Comment by mbroadst** [comment]
@daprahamian it looks like the tests are still in-flight, should we hold off on final review here until those are complete?

**Comment by daprahamian** [comment]
@mbroadst yeah, we can hold off until the spec is complete

File: lib/change_stream.js [hunk 2]
```diff
@@ -296,42 +297,41 @@ ChangeStream.prototype.stream = function(options) {
...
+  11601 // Interrupted
 ]);
 
-// TODO: will be used for check for getMore errors
-// const GET_MORE_NON_RESUMABLE_CODES = new Set([
-//   136, // CappedPositionLost
-//   237, // CursorKilled
-//   11601 // Interrupted
-// ]);
+ChangeStream.GET_MORE_NON_RESUMABLE_CODES = GET_MORE_NON_RESUMABLE_CODES;
```
Comment (mbroadst) [inline lib/change_stream.js]: why export this?

File: test/unit/change_stream_resume_tests.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+const expect = require('chai').expect;
+const mock = require('mongodb-mock-server');
+const ChangeStream = require('../../lib/change_stream');
+const MongoClient = require('../../lib/mongo_client');
+const ObjectId = require('../../index').ObjectId;
+const Timestamp = require('../../index').Timestamp;
+const Long = require('../../index').Long;
+
+describe.only('Change Stream Resume Tests', function() {
```
Comment (mbroadst) [inline test/unit/change_stream_resume_tests.js]: forgot to remove the `.only` here

File: test/unit/change_stream_resume_tests.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  const test = {};
+  const DEFAULT_IS_MASTER = {
```
Comment (mbroadst) [inline test/unit/change_stream_resume_tests.js]: https://github.com/mongodb-js/mongodb-mock-server/blob/master/index.js#L74

File: test/unit/change_stream_resume_tests.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+  let changeStream;
+
+  beforeEach(() => {
+    return mock.createServer().then(server => {
+      test.server = server;
+    });
+  });
+  afterEach(done => changeStream.close(() => client.close(() => mock.cleanup(done))));
+
+  configs.forEach(config => {
```
Comment (mbroadst) [inline test/unit/change_stream_resume_tests.js]: nice test!

**Pull request merged**